### ISSUE TITLE
Can't ALT-TAB out of full-screen process when hooked

### ISSUE THREAD

**Issue by benjaminmordaunt** [Rebzzel/kiero#29]
Context:
I am using DX9 + imGUI + kiero + DLL injection.
Also, I have implemented the recommended hkReset hook for when the context resets/changes resolution/enters/exits full-screen.

This issue is broken into two parts:

1. If you inject the DLL & bind methods when the DirectX process is full-screen, the imGUI interface doesn't appear. However, if you inject the module when in windowed mode, then maximise or enter full-screen mode, the interface appears as expected.

2. When the DLL is injected and the DirectX process is in full-screen, you cannot ALT-TAB or use the Windows key to exit the process. ALT-TAB allows you to select a target window, but does not change focus to that new application.

Please let me know what additional data you may need, thanks.

**Comment by Rebzzel** [comment]
Hello, @crawben20!

Thank you for reporting this bug!
 
Have you checked what kiero returns when ImGui doesn't show?
I'll fix when I have some free time. 
I think something is wrong with CreateDevice.

Have a nice day!

**Comment by Rebzzel** [comment]
Yeah. I spent a day trying to fix this bug.

After searching the Internet, I noticed that those who used a similar method of hook (aka. "dummy device hook"), experienced this bug, but no one ever answered them right how to fix it.

As I expected, the bug was in CreateDevice. 
Since I didn't find any useful information about this, I had to go into the d3d9.dll and watch the source code.
At first I thought the problem was in hFocusWindow, but when analyzing the code, I didn't find any possible issues. 
I was desperate, but decided to change the other parameters of the function, and in the end "BINGO! WORKING!".

So, we can say we are the first who were able to fix this bug (at least in the public).
